esy-installer/dune

```diff
@@ -0,0 +1,6 @@
+(library
```

Comment:
I'm seeing build failures on Windows, I think we'll need to hook up this new library to our 'bootstrap' build scripts. TH e failures are:
```
dune:build esy-build-package
File "esy-build-package/bin/dune", line 6, characters 14-27:
Error: No ppx driver were found.
Hint: Try upgrading or reinstalling ocaml-migrate-parsetree and ppxlib.
File "esy-build-package/dune", line 5, characters 14-64:
Error: No ppx driver were found.
Hint: Try upgrading or reinstalling ocaml-migrate-parsetree and ppxlib.
File "esy-installer/dune", line 4, characters 14-27:
Error: No ppx driver were found.
Hint: Try upgrading or reinstalling ocaml-migrate-parsetree and ppxlib.
File "esy-lib/dune", line 4, characters 14-72:
Error: No ppx driver were found.
Hint: Try upgrading or reinstalling ocaml-migrate-parsetree and ppxlib.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! esy@0.2.0 bootstrap:build: `esy-bash ./scripts/bootstrap/build-bootstrap.sh`
npm ERR! Exit status 1
```

Reply:
I googled this and it seems because of a stale opam cache on appveyor (it passed on another push build in the prev PR which converted to dune). I’ll take a look on how to invalidate appveyor cache.

Reply:
Ah ok - good find. I was having trouble repro'ing locally because I had to re-install my cache with the dune changes.

Here's info on clearing the cache:
https://www.appveyor.com/docs/build-cache/#remove-cache-entry-from-build-config

Travis is nice because it lets you explicitly delete the cache. For now, we can just comment out the `cache` section in our appveyor.yml at the root, and bring it back after this change.

Reply:
One thing could also look at is use AppVeyor's cache invalidation, like:

`C:\Users\appveyor\.opam  -> **\dune` <-- invalidate the cache whenever a `dune` file changes.

Reply:
Validated after clearing / re-installing my OPAM repo locally that the build succeeds. I opened #292 to clear the cache in cases where `dune` files change.

Reply:
Alright, merged #292 and rebased on top of, trying.